### PR TITLE
Fix escaping for `cmd.exe`: either quote or escape individual special characters

### LINKED ISSUE #46: Escaping for `cmd.exe` unnecessarily escapes parenthesis
The path escaping for `cmd.exe` turns

```
"C:\Program Files (x86)\R\R-4.3.3\bin\Rscript.exe"
```

into

```
"C:\Program Files ^(x86^)\R\R-4.3.3\bin\Rscript.exe"
#> The system cannot find the path specified.
```

Testing with Windows 11 via Parallels

### PR DESCRIPTION
**Pull request by gadenbuie**
Fixes #46 

Prior to this PR, for `cmd.exe` commands, we would escape special characters and then, if the escaped string contains a space we'd wrap the string in `" "`.

For a string like `C:\Program Files (x86)\R\R-4.3.3\bin\Rscript.exe`, we'd turn `(` and `)` into `^(` and `^)`, but we'd then also wrap the string in `""` resulting in 

```
"C:\Program Files ^(x86^)\R\R-4.3.3\bin\Rscript.exe"
```

This would break because the `""` handles the escaping of the parenthesis. This PR changes the escaping to either escape individual special characters or wrap the string in `""` if it contains a space.

### CODE REVIEW

File: src/shell-utils.ts [hunk 1]
```diff
@@ -33,8 +33,9 @@ export function escapeArg(arg: string, style: EscapeStyle): string {
   switch (style) {
     case "cmd":
       // For cmd.exe, double quotes are used to handle spaces, and carets (^) are used to escape special characters.
-      const escaped = arg.replace(/([()%!^"<>&|])/g, "^$1");
-      return /\s/.test(escaped) ? `"${escaped}"` : escaped;
+      return /\s/.test(arg)
```
Comment (jcheng5) [inline src/shell-utils.ts:36]: I think the caret approach needs to be used if the argument contains a double quote. It would be more readable to switch to single quotes, unless the argument contained both types of quotes... but I think this suggestion should be correct for every case and is readable enough.

(I have not actually tried it though)
```suggestion
      return (/\s/.test(arg) && !/"/test(arg))
```

**Comment by jcheng5** [comment]
Phew, good catch

File: src/shell-utils.ts (hunk 1 above)
Comment (jcheng5) [reply src/shell-utils.ts]: Yeah I realize it's a super edge case but I'm being extra nitpicky about it because 1) this is pretty much a general purpose escape function even if we're only currently using it for this one purpose, and 2) buggy edge cases in escaping code feels like asking for an eventual security vulnerability. Better just to be bulletproof if it's not a big cost.

In my testing, echo behaves differently than everything else. To test, I created a batch file called `a.bat` with the following contents:
```
@echo off
echo %1
```
then called `a "blah"`, for example. And for me, `echo "blah"` printed `"blah"` while `a "blah"` printed `blah`.

Comment (gadenbuie) [reply src/shell-utils.ts]: Thanks, I agree with the reasons for the nitpicking. I'm not sure if you saw -- I realized that echo's escaping is different from other functions (that seems to be a theme in Windows) and implemented the doubling of `"` when a string with a space contains a double quote.

**Review by jcheng5 — APPROVED** [review]
